Packs/SecurityScorecard/Integrations/SecurityScorecard/SecurityScorecard.py

```diff
@@ -0,0 +1,1203 @@
+
+        most_recent_alert_timestamp = \
+            int(datetime.strptime(most_recent_alert_created_date, SECURITYSCORECARD_DATE_FORMAT).timestamp())  # type: ignore
+        demisto.debug(f"Setting last runtime as alert most recent timestamp: {most_recent_alert_timestamp}")
+        demisto.setLastRun({
+            'last_run': most_recent_alert_timestamp
+        })
+
+        for alert in alerts:
+
+            alert_created_at = alert.get("created_at")
+            alert_timestamp = int(datetime.strptime(alert_created_at, SECURITYSCORECARD_DATE_FORMAT).timestamp())  # type: ignore
+
+            alert_id = alert.get("id")
+            company_name: str = alert.get("company_name")  # type: ignore
+            change_type: str = alert.get("change_type")  # type: ignore
+
+            debug_msg = f"import alert '{alert_id}'? (last_run < alert_timestamp): {(last_run < alert_timestamp)}"  # type: ignore
+
+            demisto.debug(debug_msg)
+
+            if alert_timestamp > last_run:  # type: ignore
+                incident = {}
+                incident["name"] = f"{company_name} {change_type.replace('_', ' ').title()}"
+                incident["occurred"] = \
+                    datetime.strptime(alert_created_at, SECURITYSCORECARD_DATE_FORMAT).strftime(DATE_FORMAT)  # type: ignore
+                incident["rawJSON"] = json.dumps(alert)
+                incidents_to_import.append(incident)
+    # If there are no alerts then we can't use the most recent alert timestamp
+    # So we'll use now as the last run timestamp
+    else:
+        now = int(datetime.utcnow().timestamp())
+        demisto.debug(f"No alerts retrieved, setting last_run to now ({now})")
+        demisto.setLastRun({
+            'last_run': now
+        })
+
+    return incidents_to_import
+
+
+""" COMMAND FUNCTIONS """
+
+
+def test_module(
+    client: SecurityScorecardClient,
+    incident_fetch_interval: Optional[str]
+) -> str:
+    """Tests API connectivity and authentication
+
+    Runs the fetch-alerts mechanism to validate all integration parameters
+
+    Args:
+        ``client`` (``SecurityScorecardClient``): SecurityScorecard client
+
+    Returns:
+        ``str``: 'ok' if test passed, anything else will fail the test.
+    """
+    demisto.debug("Initialized test module...")
+
+    interval = arg_to_number(arg=incident_fetch_interval, arg_name="incident_fetch_interval", required=False)
+
+    if interval > 1440 * 2:  # type: ignore
+        return "Test failed. Incident Fetch Interval is greater than 2 days."
+
+    max_incidents = int(client.max_fetch)
+    if max_incidents > 50:
+        return "Test failed. Max Fetch is larger than 50."
+
+    try:
+        client.fetch_alerts(page_size=1, page=1)
+        demisto.debug("Test module successful")
+        return('ok')
+    except DemistoException as e:
+        if 'Unauthorized' in str(e):
+            return('Authorization Error: make sure API Key is correctly set')
```

Comment:
This check should be for every command not only test module

Reply:
Being handled in `http_request` wrapper static method `error_handler`.

Reply:
Agreed, makes more sense to handle this using the `error_handler`. But why not adding this logic specifically inside this hadler function?

Reply:
> Being handled in `http_request` wrapper static method `error_handler`.

![image](https://user-images.githubusercontent.com/85439776/136698297-b95ed212-7d0e-4a5e-9a93-6caee1f0a1cd.png)